org.spoofax.terms/src/org/spoofax/terms/StringTermReader.java

```diff
@@ -236,11 +236,11 @@ private String parseDigitSequence(PushbackStringIterator bis) throws ParseError
         do {
             sb.append((char)ch);
```

Comment:
If you check that the `int ch` is a digit with `Character.isDigit`, why do you then need to cast to `char` and `append`? Does `Character.isDigit` only return `true` for codepoints that fit inside `char`?

Reply:
I missed this case, thanks for the catch! :slightly_smiling_face: To answer your question, `Character.isDigit` returns `true` for "many other character ranges": https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#isDigit(int)

Will fix!

Reply:
In that case you may want to only support ASCII digits for now, otherwise you're changing the ATerm format too much I think.

Reply:
Then probably the same holds for calls to `Character.isLetter` (for the first character of a constructor) and `Character.isLetterOrDigit` (for other characters of a constructor)? These methods also return `true` for letters from other alphabets; I guess we want to restrict this to `[0-9A-Za-z]`?

Reply:
Hmm. Yes, let's be conservative and restrict those to ASCII for now as well.